### PR TITLE
Support eventually consistent scenarios

### PR DESCRIPTION
**Pull request by carl-hartshorn**
Initial implementation of point 3 from issue #2

### CODE REVIEW

File: GherkinSpec.DogFoodTests/GherkinSpec.DogFoodTests.csproj [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (carl-hartshorn) [inline GherkinSpec.DogFoodTests/GherkinSpec.DogFoodTests.csproj]: @DaveAurionix if we want to keep these tests I'll update the build yml to run them

File: docs/EventuallyConsistentServices.md [hunk 2]
```diff
@@ -4,21 +4,42 @@ A message-based architecture (where messages are pushed onto queues and pulled o
...
+* A test exists for this behaviour: "... When event X is received, Then event Y is published."
+* Step level eventual consistency may be applied on the Then step in this test to cause the Then step to be retried until the event has been published.
+
+The intended usage of scenario level eventual consistency is as follows:
+
+* A service accepts data via HTTP PUT. The service enqueues a command to process that data to immediately return to the caller.  The service then serves the processed data via HTTP GET.
+* A test exists for this behaviour: "Given I have submitted data, When I retrieve that data, Then the data matches my expectation."
+* Scenario level eventual consistency may be applied to this test scenario to cause the When and Then steps to be retried until the data has been processed and is being served via HTTP GET.
+
+Both levels of eventual consistency may be combined, as described in the following example:
```
Comment (carl-hartshorn) [inline docs/EventuallyConsistentServices.md]: @DaveAurionix do you think this is too confusing? If so we could remove this and the related gotcha below

**Comment by carl-hartshorn** [comment]
`Roadmap.md` will need updating and the minor version incrementing when this feature is completed

File: GherkinSpec.DogFoodTests/GherkinSpec.DogFoodTests.csproj (hunk 1 above)
Comment (carl-hartshorn) [reply GherkinSpec.DogFoodTests/GherkinSpec.DogFoodTests.csproj]: Removed as discussed

File: docs/EventuallyConsistentServices.md (hunk 2 above)
Comment (DaveAurionix) [reply docs/EventuallyConsistentServices.md]: As discussed, we'll remove this.

Comment (carl-hartshorn) [reply docs/EventuallyConsistentServices.md]: Removed as discussed

**Pull request merged**